compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3092,9 +3324,55 @@ abstract class GenJSCode extends plugins.PluginComponent
       val receiver = genExpr(receiver0)
       val args = genPrimitiveJSArgs(sym, args0)
 
+      genJSCallGeneric(sym, receiver, args, isStat)
+    }
+
+    private def genJSSuperCall(tree: Apply, isStat: Boolean): js.Tree = {
+      implicit val pos = tree.pos
+      val Apply(fun @ Select(sup @ Super(_, mix), _), args) = tree
+      val sym = fun.symbol
+
+      val genReceiver = genThis()(sup.pos)
+      lazy val genScalaArgs = genActualArgs(sym, args)
+      lazy val genJSArgs = genPrimitiveJSArgs(sym, args)
+
+      if (sym.owner == ObjectClass) {
+        // Normal call anyway
+        assert(!sym.isClassConstructor,
+            "Trying to call the super constructor of Object in a " +
+            s"Scala.js-defined JS class at $pos")
+        genApplyMethod(genReceiver, sym, genScalaArgs)
+      } else if (sym.isClassConstructor) {
+        // Implementation restriction
+        if (genJSArgs.exists(_.isInstanceOf[js.JSSpread])) {
+          reporter.error(pos,
+              "Implementation restriction: cannot call the super " +
+              "constructor of a Scala.js-defined JS class with :_*")
+        }
+
+        js.JSSuperConstructorCall(genJSArgs)
+      } else if (isScalaJSDefinedJSClass(sym.owner) && !isExposed(sym)) {
+        // Reroute to the static method
+        genApplyJSClassMethod(genReceiver, sym, genScalaArgs)
+      } else {
+        genJSCallGeneric(sym, genReceiver, genJSArgs, isStat,
+            superIn = Some(currentClassSym))
```

Comment:
> When do we hit this case?

Most of the time. Almost every time you call `super` in a Scala.js-defined JS class.
```scala
class NativeParent extends js.Object {
  def someGetter: Int = js.native
  def foo(x: Int): Int = js.native
  def apply(x: String): String = js.native
}

@ScalaJSDefined
class ScalaJSDefinedChild extends NativeParent {
  override def someGetter: Int = super.someGetter + 3
  override def foo(x: Int): Int = super.foo(x) + 1
  def foobar(x: Int): Int = super.foo(x) * 2
  def bar(x: String): String = super.apply(x)
}
```
All lines in `ScalaJSDefinedChild` end up in this case. Also applies if `NativeParent` is in fact non native. The fourth one hits the special case for `apply` in `genJSCallGeneric`, which will cause a compile-error. You can also call `super` on bracket-select and bracket-call symbols. `genJSCallGeneric` captures all of that.